components/function-controller/cmd/manager/main.go

```diff
@@ -70,25 +67,41 @@ func main() {
 		os.Exit(1)
 	}
 
-	logLevel, err := toZapLogLevel(config.LogLevel)
+	logLevel, err := logger.MapLevel(config.LogLevel)
 	if err != nil {
 		ctrl.SetLogger(ctrlzap.New())
 		setupLog.Error(err, "unable to set logging level")
 		os.Exit(2)
 	}
+	format, err := logger.MapFormat(config.LogFormat)
+	if err != nil {
+		ctrl.SetLogger(ctrlzap.New())
+		setupLog.Error(err, "unable to set logging format")
+		os.Exit(1)
+	}
 
-	atomicLevel := zap.NewAtomicLevelAt(logLevel)
-	zapLogger := ctrlzap.NewRaw(ctrlzap.Level(&atomicLevel))
-	ctrl.SetLogger(zapr.NewLogger(zapLogger))
+	l, err := logger.New(format, logLevel)
+	if err != nil {
+		ctrl.SetLogger(ctrlzap.New())
+		setupLog.Error(err, "unable to set setup logger level")
+		os.Exit(3)
+	}
+
+	if err := logger.InitKlog(l, logLevel); err != nil {
+		setupLog.Error(err, "unable to set setup logger level")
+		os.Exit(5)
+	}
+	ctrl.SetLogger(zapr.NewLogger(l.WithContext().Desugar()))
```

Comment:
I think we can should wrap the log setup in a function. I would probably be useful when we apply the same to the webhook as well.